gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala

```diff
@@ -607,10 +607,32 @@ trait HashJoinLikeExecTransformer
     (right, left)
   }
 
+  def sameType(from: DataType, to: DataType): Boolean = {
```

Comment:
Spark has already provided a method called `sameType` in `DataType.scala`. I think it would be better to call spark's method, which will reduce our effort in maintaining such fundamental method.
I note `sameType` method in spark is package private. Maybe, we can create a util class in a same named package to make this spark method accessible. Then, we can call util class's corresponding method here. Right?

Reply:
The the 'sameType' is not accessible because of private package constraint.
Also will make many adjustments to use this method. So instead, created a rename function to use.